### PR TITLE
[BetterPhpDocParser] Add ArrayItemNode to improve value transfer in annotation curly lists

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: tests/Issues/PartialValueDocblockUpdate/Fixture/add_default_value_to_route.php.inc [hunk 1]
```diff
@@ -1,42 +1,38 @@
...
 use Symfony\Component\Routing\Annotation\Route;
 
-class Test {
+final class AddDefaultValueToRoute
+{
     /**
      * @Since("6.4.0.1")
      * @Route("/api/_admin/reset-excluded-search-term", name="api.admin.reset-excluded-search-term", methods={"POST"})
-     *
-     * @return JsonResponse
```
Comment (samsonasik) [inline tests/Issues/PartialValueDocblockUpdate/Fixture/add_default_value_to_route.php.inc:13]: No, this `@return` here is to reproduce the issue, and  the order need to stay

Comment (samsonasik) [reply tests/Issues/PartialValueDocblockUpdate/Fixture/add_default_value_to_route.php.inc:13]: see https://github.com/rectorphp/rector-src/pull/1862 for why it is needed for reproduce the error

Comment (TomasVotruba) [reply tests/Issues/PartialValueDocblockUpdate/Fixture/add_default_value_to_route.php.inc:13]: I don't see any exact reason for the `@return` to be there. It's only in original fixture.

This should test the Doctrine annotation is changed propperly.

Comment (samsonasik) [reply tests/Issues/PartialValueDocblockUpdate/Fixture/add_default_value_to_route.php.inc:13]: Yes, but without the `@return JsonResponse`, the original issue was never exists, it was issue with SpacelessPhpDocTagNode next transformation

Comment (TomasVotruba) [reply tests/Issues/PartialValueDocblockUpdate/Fixture/add_default_value_to_route.php.inc:13]: I see. I'll add it back then

File: rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/Doctrine/preserve_int_key_defined.php.inc [hunk 2]
```diff
@@ -19,7 +22,7 @@ namespace Rector\Tests\Php80\Rector\Class_\AnnotationToAttributeRector\Fixture\D
 
 use Doctrine\ORM\Mapping as ORM;
 
-#[ORM\DiscriminatorMap([1 => 'CostDetailEntity'])]
+#[ORM\DiscriminatorMap(['cost' => 'CostEntity', 'product' => 'ProductEntity'])]
```
Comment (samsonasik) [inline rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/Doctrine/preserve_int_key_defined.php.inc:25]: "1" to 1 need to be kept to represent fixture to keep integer key unordered (direct 1)

Comment (samsonasik) [reply rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/Doctrine/preserve_int_key_defined.php.inc:25]: see the original PR why key 1 added here https://github.com/rectorphp/rector-src/pull/2735

Comment (TomasVotruba) [reply rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/Doctrine/preserve_int_key_defined.php.inc:25]: I checked this fixture and this is always `array<string, string>` list:
https://www.doctrine-project.org/projects/doctrine-orm/en/latest/reference/inheritance-mapping.html#class-table-inheritance

Not sure how this got through :)

Comment (samsonasik) [reply rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/Doctrine/preserve_int_key_defined.php.inc:25]: the issue show the usage of key 1 directly https://github.com/rectorphp/rector/issues/7345

Comment (TomasVotruba) [reply rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/Doctrine/preserve_int_key_defined.php.inc:25]: I need to finish the feature first, then we'll deal with those edgecases. Otherwise I'm loosing focus and cannot progress.

**Comment by TomasVotruba** [comment]
CI passed :tada:  :tada: :tada: 

Let's ship it to test edge-cases :) this will make changing particular items much easier, with no quote/double quote/single quote magic :+1:

**Comment by samsonasik** [comment]
@TomasVotruba I create new PR to preserve key unordered not from 0 integer rollback https://github.com/rectorphp/rector-src/pull/2811
